src/main/java/com/nha/abdm/wrapper/hip/hrp/link/hipInitiated/HipLinkService.java

```diff
@@ -170,21 +167,35 @@ public void hipConfirmCall(LinkOnInitResponse linkOnInitResponse) {
             .dateOfBirth(patient.getDateOfBirth())
             .build();
     LinkCredential linkCredential = LinkCredential.builder().demographic(userDemographic).build();
-    LinkConfirm linkConfirm =
-        LinkConfirm.builder()
+    LinkConfirmRequest linkConfirmRequest =
+        LinkConfirmRequest.builder()
             .requestId(UUID.randomUUID().toString())
             .timestamp(Utils.getCurrentTimeStamp())
             .transactionId(linkOnInitResponse.getAuth().getTransactionId())
             .credential(linkCredential)
             .build();
-    log.debug("hipConfirmCall" + linkConfirm.toString());
-    requestLogService.setHipOnInitResponse(linkOnInitResponse, linkConfirm);
+    log.debug("confirmAuthDemographics linkConfirmRequest: " + linkConfirmRequest.toString());
+    requestLogService.updateHipOnInitResponse(linkOnInitResponse, linkConfirmRequest);
     try {
-      responseEntity =
-          requestManager.fetchResponseFromPostRequest(linkConfirmAuthPath, linkConfirm);
-      log.info(linkConfirmAuthPath + " : linkConfirmAuth: " + responseEntity.getStatusCode());
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(linkConfirmAuthPath, linkConfirmRequest);
+      log.info(linkConfirmAuthPath + " : confirmAuthDemographics: " + response.getStatusCode());
+      if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+        requestLogService.updateStatus(requestLog, RequestStatus.AUTH_CONFIRM_ACCEPTED);
+      } else if (Objects.nonNull(response.getBody())) {
+        requestLogService.updateError(
```

Comment:
The error is getting saved here which is good, but it would be good to update the corresponding RequestStatus to an error state which represents the current state.

Reply:
made changes, thanks